File: backend/app/schemas/ai_settings.py
from datetime import datetime
from typing import Literal

from pydantic import UUID4, BaseModel, Field

ALLOWED_AI_PROVIDERS = {"openai", "anthropic", "ollama", "lmstudio"}


class AISettingsProfile(BaseModel):
    id: UUID4
    provider: str | None = None
    model: str | None = None
    base_url: str | None = None
    gateway_route: str | None = None
    updated_at: datetime | None = None


class AISettingsEffective(BaseModel):
    provider: str
    model: str
    base_url: str | None = None
    gateway_route: str | None = None
    mode: str


class AISettingsRead(BaseModel):
    profile: AISettingsProfile | None = None
    effective: AISettingsEffective


class AISettingsUpdate(BaseModel):
    provider: str | None = Field(default=None, max_length=50)
    model: str | None = Field(default=None, max_length=200)
    base_url: str | None = Field(default=None, max_length=500)
    gateway_route: str | None = Field(default=None, max_length=200)


class AISettingsTestInput(BaseModel):
    provider: str | None = Field(default=None, max_length=50)
    model: str | None = Field(default=None, max_length=200)
    base_url: str | None = Field(default=None, max_length=500)
    gateway_route: str | None = Field(default=None, max_length=200)


class AISettingsTestResult(BaseModel):
    status: Literal["ok", "error"]
    latency_ms: int | None = None
    model: str | None = None
    message: str | None = None

File: frontend/src/modules/ai-settings/models/aiSettings.ts
export type AIProvider = 'openai' | 'anthropic' | 'ollama' | 'lmstudio'

export type AIMode = 'gateway' | 'direct' | 'ollama' | 'lmstudio' | 'disabled'

export interface AIProfile {
  id: string
  provider: AIProvider | null
  model: string | null
  base_url: string | null
  gateway_route: string | null
  updated_at: string | null
}

export interface AIEffective {
  provider: string
  model: string
  base_url: string | null
  gateway_route: string | null
  mode: AIMode
}

export interface AISettingsRead {
  profile: AIProfile | null
  effective: AIEffective
}

export interface AISettingsUpdate {
  provider?: AIProvider | null
  model?: string | null
  base_url?: string | null
  gateway_route?: string | null
}

export const AI_PROVIDER_OPTIONS: { value: AIProvider | ''; label: string }[] = [
  { value: '', label: 'Default (from env)' },
  { value: 'openai', label: 'OpenAI' },
  { value: 'anthropic', label: 'Anthropic' },
  { value: 'ollama', label: 'Ollama' },
  { value: 'lmstudio', label: 'LM Studio' },
]

export type AISettingsTestResult =
  | { status: 'ok'; latency_ms: number; model: string }
  | { status: 'error'; latency_ms: number | null; model: string | null; message: string }
